s3fs/core.py

```diff
@@ -509,9 +509,15 @@ def _open(
             requester_pays=requester_pays,
         )
 
-    async def _lsdir(self, path, refresh=False, max_items=None, delimiter="/"):
+    async def _lsdir(
+        self, path, refresh=False, max_items=None, delimiter="/", use_prefix=True
+    ):
         bucket, prefix, _ = self.split_path(path)
-        prefix = prefix + "/" if prefix else ""
+        if prefix:
+            if use_prefix:
+                prefix += "/"
```

Comment:
The name of the argument "use_prefix" seems to have the opposite meaning to its function - at least to how it seems to me. The current default, which matches previous behaviour, is the right one, though.

Reply:
Yeah, I don't quite like it as well. That was the first to come to my mind (use_prefix => use the forward slash in the `Prefix=...`).  Any suggestions?